internal/pkg/cli/task_run.go

```diff
@@ -138,7 +154,59 @@ func (o *runTaskOpts) Ask() error {
 	return nil
 }
 
+// Execute deploys and runs the task.
 func (o *runTaskOpts) Execute() error {
+	if err := o.deployTaskResources(); err != nil {
+		return err
+	}
+	// NOTE: if image is not provided, then we build the image and push to ECR repo
+	if o.image == "" {
+		if err := o.buildAndPushImage(); err != nil {
+			return err
+		}
+
+		if err := o.updateTaskResources(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o *runTaskOpts) deployTaskResources() error {
+	o.spinner.Start(fmt.Sprintf("Deploying resources for task %s", color.HighlightUserInput(o.groupName)))
```

Comment:
I feel like it would be nice for users to get a sense of the type of resources that we need to create to run the task. What do you think of?
```suggestion
	o.spinner.Start(fmt.Sprintf("Creating a task definition for %s and the necessary permissions", color.HighlightUserInput(o.groupName)))
```

Reply:
Then should I also mention the ECR repo and log group that are also created?

Reply:
I wasn't sure about that in case this step only did a stack update instead of create 🤔 
Do you know which resource takes the longest to create, I feel like as long as we mention that one the rest shouldn't matter as much 8)

Reply:
In fact, neither of the stack resource (roles, ecr repo, log group) takes long to provision. From CloudFormation console, each of them takes about 1-2seconds to create lol. 

Task definition is not created at the first call; it's only created/updated after the image is pushed, i.e. the second call. During the second call, it takes ~2 seconds to create the task definition - according to CloudFormation console.

So most of the waiting is spent on building and pushing images, and waiting for tasks to start.

Reply:
> I wasn't sure about that in case this step only did a stack update instead of create 🤔

We can perhaps use the word 'Preparing' instead of 'Creating' or 'Deploying', if we want to include ECR/log group.

Reply:
Does it make sense to use the term `Provisioning resources for task %s` instead of preparing?